src/triage/tracking.py

```diff
@@ -0,0 +1,282 @@
+
+
+def infer_git_hash():
+    """Attempt to infer the git hash of the repository in the current working directory
+
+    Returns: Either the 'git rev-parse HEAD' output or None
+    """
+    try:
+        git_hash = subprocess.check_output(['git', 'rev-parse', 'HEAD']).strip().decode('utf-8')
+    except Exception as e:
+        logging.info("Unable to infer git hash, error: %s", str(e))
+        git_hash = None
+    return git_hash
+
+
+def infer_triage_version():
+    return __version__
+
+
+def infer_installed_libraries():
+    """Attempt to infer the installed libraries by running pip freeze and formatting as a list
+
+    Returns: Either a list, or None
+    """
+    try:
+        installed_libraries = subprocess.check_output(['pip', 'freeze']).split(b'\n')
+    except FileNotFoundError:
+        logging.info("Unable to pip freeze, cannot list installed libraries")
+        installed_libraries = []
+    return installed_libraries
+
+
+def infer_ec2_instance_type():
+    """Attempt to infer the instance type of the ec2 instance by querying Amazon's endpoint
+
+    Returns: Either the ec2 instance type as returned by Amazon's endpoint, or None
+    """
+    try:
+        ec2_instance_type = requests.get(
+            'http://169.254.169.254/latest/meta-data/instance-type',
+            timeout=1
+        ).text
+    except requests.exceptions.ConnectionError:
+        logging.info(
+            "Unable to retrieve metadata about ec2 instance, will not set ec2 instance type"
+        )
+        ec2_instance_type = None
+    return ec2_instance_type
+
+
+def infer_log_location():
+    """Attempt to infer the location of the log file of the root logger
+
+    Returns: Either the baseFilename of the first FileHandler on the root logger, or None
+    """
+    root_logger_handlers = [
+        handler
+        for handler in logging.getLoggerClass().root.handlers
+        if isinstance(handler, logging.FileHandler)
+    ]
+    if root_logger_handlers:
+        log_location = root_logger_handlers[0].baseFilename
+    else:
+        logging.info("No FileHandler found in root logger, cannot record logging filename")
+        log_location = None
+    return log_location
+
+
+def initialize_tracking_and_get_run_id(
+    experiment_hash,
+    experiment_class_name,
+    experiment_kwargs,
+    db_engine
+):
+    """Create a row in the ExperimentRun table with some initial info and return the created run_id
+
+    Args:
+        experiment_hash (str) An experiment hash that exists in the experiments table
+        experiment_class_name (str) The name of the experiment subclass used
+        experiment_kwargs (dict) Any runtime Experiment keyword arguments that should be saved
+        db_engine (sqlalchemy.engine)
+    """
+    # Any experiment kwargs that are types (e.g. MatrixStorageClass) can't
+    # be serialized, so just use the class name if so
+    cleaned_experiment_kwargs = {
+        k: (v.__name__ if isinstance(v, type) else v)
```

Comment:
again, would be nicer to include the class path; (this can be a simple helper if needed).

it's basically all classes' `repr`, but I don't see a pre-existing nice method to get it. easy enough for us to construct.